tempesta_fw/ss_skb.c

```diff
@@ -1021,6 +1021,20 @@ ss_skb_queue_coalesce_tail(SsSkbList *skb_list, const struct sk_buff *skb)
 	return 0;
 }
 
+static inline void
+__copy_ip_header(struct sk_buff *to, const struct sk_buff *from)
+{
+	const struct iphdr *ip4 = ip_hdr(from);
+	const struct ipv6hdr *ip6 = ipv6_hdr(from);
+
+	if (ip6->version == 6)
+		memcpy(to->data, ip6, sizeof(*ip6));
+	else
+		memcpy(to->data, ip4, sizeof(*ip4));
+
+	skb_reset_network_header(to);
+}
```

Comment:
Why do we need to copy whole IP header? Secondly, if it's done for `tfw_http_add_x_forwarded_for()` only, then probably we can just add IP address field to TfwHttpMsg and initialize it with first skb, then `tfw_http_add_x_forwarded_for()` can take address from there.